src/coins.cpp

```diff
@@ -379,6 +382,24 @@ CSidechainEventsMap::iterator CCoinsViewCache::ModifySidechainEvents(int height)
     return ret;
 }
 
+CCertDataHashMap::const_iterator CCoinsViewCache::FetchCertDataEntry(const uint256& scId, const int epoch) const {
+    auto key = std::make_pair(scId, epoch);
+    CCertDataHashMap::iterator candidateIt = cacheCertDataHashes.find(key);
+    if (candidateIt != cacheCertDataHashes.end())
+        return candidateIt;
+
+    std::pair<libzendoomc::ScFieldElement, libzendoomc::ScFieldElement> baseData;
+    if (!base->GetCertDataHashes(scId, epoch, baseData))
+        return cacheCertDataHashes.end();
+
+    //Fill cache and return iterator. The insert in cache below looks cumbersome. However
+    //it allows to insert CCeasingSidechains and keep iterator to inserted member without extra searches
+    CCertDataHashMap::iterator ret = cacheCertDataHashes.insert(std::make_pair(key, CCertDataHashCacheEntry(baseData, CCertDataHashCacheEntry::Flags::DEFAULT ))).first;
+
+    //cachedCoinsUsage += ret->second.scEvents.DynamicMemoryUsage(); //TODO
```

Comment:
Do we need this one here?

Reply:
We need to account for the DynamicMemoryUsage of CertDataHashes. Going to fix this in a different PR